### PR TITLE
Report Command

### PR DESCRIPTION
**Pull request by GamingMineblox**
**Please describe the changes this pull request makes and why it should be merged:**

> I have made an ***setreport {#channel / id}** command and a ***report {@user} {reason}** from the suggestion from `mouton1234#3895`:
image

**Status:**

- [x] Code changes have been tested
- [x] Code changes work without errors

**Content of the pull request:**  

- [x] This pull request changes the bot
  - [ ] This pull request includes breaking changes for the bot
  - [x] This pull request includes new command(s) for the bot

- [x] This pull request changes the dashboard

- [ ] This pull request **only** includes non-code changes, like changes to templates, README.md, languages, etc.

### CODE REVIEW

**Comment by Androz2091** [comment]
Very good job 👏 
Can you update the french language too?

```Js
/* REPORT COMMAND */
REPORT_DESCRIPTION: "Envoyez votre signalement dans le salon défini pour cela !",
REPORT_USAGE: "report [@user] [raison]",
REPORT_EXAMPLES: "$report @Androz#2091 Règles non respectées",
// Errors
REPORT_ERR_NO_CHANNEL: `${e.error} | Aucun salon de signalement défini !`,
REPORT_ERR_NO_REP: `${e.error} | Veuillez entrer une raison pour votre signalement !`,
REPORT_ERR_NO_USER: `${e.error} | Veuillez mentionner l'utilisateur à signaler !`,
//Headings
REPORT_HEADINGS: [
	"Auteur",
	"Date",
	"Contenu",
	"Signalement"
],
// Content
REPORT_TITLE: (user) => `Signalement - ${user.tag}`,
REPORT_SUCCESS: (channel) => `${e.success} | Votre report est publié dans ${channel} !`,
```

File: languages/english.js [hunk 1]
```diff
@@ -385,6 +385,25 @@ module.exports = class {
 			EVAL_USAGE: "eval [code]",
 			EVAL_EXAMPLES: "$eval message.channel.send('Hey');",
 
+			/* REPORT COMMAND */
+			REPORT_DESCRIPTION: "Send your report to the channel defined for this!",
+			REPORT_USAGE: "report [@user] [reason]",
+			REPORT_EXAMPLES: "$report @Androz#2091 Breaking the rules",
+			// Errors
+			REPORT_ERR_NO_CHANNEL: `${e.error} | No suggestion channel defined!`,
```
Comment (Androz2091) [inline languages/english.js]: Can you replace `suggestion` with `report`?

File: languages/english.js [hunk 2]
```diff
@@ -385,6 +385,25 @@ module.exports = class {
...
+			REPORT_ERR_NO_REP: `${e.error} | Please enter a reason`,
+			REPORT_ERR_NO_USER: `${e.error} | Please enter a user`,
```
Comment (Androz2091) [inline languages/english.js]: To give more informations, I think `Please enter a reason for your report!` and `Please mention the user you want report!` would be better.

File: languages/english.js [hunk 3]
```diff
@@ -385,6 +385,25 @@ module.exports = class {
...
+			//Headings
+			REPORT_HEADINGS: [
+				"Author",
+				"Date",
+				"Content",
+				"Reporting"
+			],
+			// Content
+			REPORT_TITLE: (user) => `Report - ${user.tag}`,
+			REPORT_SUCCESS: (channel) => `${e.success} | Your report is public in ${channel} !`,
```
Comment (Androz2091) [inline languages/english.js]: I think `Your report has been sent in ${channel} !` would be better

Comment (GamingMineblox) [reply languages/english.js]: yea you'r right

File: commands/General/report.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+            return message.channel.send(message.language.get("ERR_BOT_USER"));
+        }
+
+        let rep = args.slice(1).join(" ");
+        if(!rep){
+            return message.channel.send(message.language.get("REPORT_ERR_NO_REP"));
+        }
+        if(member.id === message.author.id){
+            return message.channel.send(message.language.get("REPORT_ERR_USER_YOURSELF"));
+        }
```
Comment (Androz2091) [inline commands/General/report.js]: I think this condition should be before the report condition.

File: commands/General/report.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+        let embed = new Discord.MessageEmbed()
+            .setAuthor(message.language.get("REPORT_TITLE", message.author), message.author.displayAvatarURL())
+            .addField(message.language.get("REPORT_HEADINGS")[0], `\`${message.author.username}#${message.author.discriminator}\``, true)
```
Comment (Androz2091) [inline commands/General/report.js]: It's better to use `message.author.tag` instead of username+#+discriminator

**Pull request merged**